subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -269,6 +269,58 @@ eclipse {
 Previously, you had to explicitly configure PMD to ignore default rules with `ruleSets = []`.
 In the Gradle 8.0, setting `ruleSetConfig` or `ruleSetFiles` to a non-empty value implicitly ignores default rules.
 
+==== Changing test framework after setting test framework options is now an error
+
+When configuring the built-in test task for Java, Groovy, and Scala projects, Gradle no longer allows you to
+change the test framework used by the `Test` task after configuring options. This was deprecated since it silently
+discarded configuration in some cases.
+
+The following code example now produces an error:
+```
+test {
+   options {
+   }
+
+   useJUnitPlatform()
+}
+```
+Instead, you can:
+
+- set the test framework before configuring options
+- migrate to the <<jvm_test_suite_plugin#jvm_test_suite_plugin,JVM Test Suite Plugin>>
+```
+test {
+   // select test framework before configuring options
+   useJUnitPlatform()
+   options {
+   }
+}
+```
+
+==== Report `getOutputLocation` return type changed
+
+In the link:{groovyDslPath}/org.gradle.api.reporting.Report.html#org.gradle.api.reporting.Report[Report] interface the return type of the `outputLocation` property is now `Property<? extends FileSystemLocation>`.  It was previously `Provider<? extends FileSystemLocation>`.
+
+This change makes the Report API more internally consistent, and allows for more idiomatic configuration of reporting tasks, using:
+
+```groovy
+tasks.named('test') {
+    reports.junitXml.outputLocation = layout.buildDirectory.dir('reports/my-report')
+}
+```
+
+versus the now `@Deprecated` former usage of:
+
+```groovy
+tasks.named('test') {
+    reports.junitXml.setDestination(layout.buildDirectory.file('reports/my-report-old').get().asFile)
+}
+
+```
```

Comment:
Consider putting the old version of the usage first or eliminating it entirely. Putting the deprecated usage after the new usage can be confusing for readers, who often expect old *before* new.

Reply:
Good idea.